lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java

```diff
@@ -200,20 +200,38 @@ public APIGatewayProxyResponseEvent handleRequest(
                     HttpStatusCode.FORBIDDEN,
                     new CommonExpressOAuthError(
                             OAuth2Error.ACCESS_DENIED, customOAuth2ErrorDescription));
+        } catch (NumberFormatException e) {
+            LOGGER.error(String.format("Parsing value of %s to int failed", MAXIMUM_ATTEMPT_COUNT));
+            eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatusCode.INTERNAL_SERVER_ERROR,
+                    new CommonExpressOAuthError(OAuth2Error.SERVER_ERROR));
         } catch (OAuthHttpResponseExceptionWithErrorBody e) {
             eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
 
-            LOGGER.debug(e.getMessage(), e);
+            // Debug only as Oauth errors appear in the redirect url
+            // This will output the specific error message
+            if (LOGGER.isDebugEnabled()) {
+                String customOAuth2ErrorDescription = e.getErrorReason();
+                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                        e.getStatusCode(), // Status Code determined by throw location
+                        new CommonExpressOAuthError(
+                                OAuth2Error.SERVER_ERROR, customOAuth2ErrorDescription));
+            }
 
+            // Non-debug route - standard OAuth2Error.SERVER_ERROR
             return ApiGatewayResponseGenerator.proxyJsonResponse(
                     e.getStatusCode(), // Status Code determined by throw location
                     new CommonExpressOAuthError(OAuth2Error.SERVER_ERROR));
-        } catch (Exception e) {
+        } catch (RuntimeException e) {
             eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
-            // Cannot log e, due to possibility of PII
-            LOGGER.error("Check Passport Unhandled Exception");
 
-            LOGGER.debug(e.getMessage(), e);
+            // This is where unexpected exceptions will reach (null pointers etc)
+            // Expected exceptions should be caught and thrown as
+            // OAuthHttpResponseExceptionWithErrorBody
+            // We should not log unknown exceptions, due to possibility of PII
+            LOGGER.error("Check Passport Unhandled Exception");
```

Comment:
As a minimum I think we should still be logging the type. Were more likely to have unknown errors we need to identify than we are to have PII currently

Reply:
Changed to `LOGGER.error("Check Passport Unhandled Exception - Type {}", e.getClass());`